babel/numbers.py

```diff
@@ -1027,11 +1029,21 @@ def parse_number(
     :raise `UnsupportedNumberingSystemError`: if the numbering system is not supported by the locale.
     """
     try:
-        return int(string.replace(get_group_symbol(locale, numbering_system=numbering_system), ''))
+        # Get the group symbol from the locale
+        group_symbol = get_group_symbol(locale, numbering_system=numbering_system)
+
+        # Replace non-breakable spaces with the group symbol
+        cleaned_string = string.replace('\xa0', group_symbol)
```

Comment:
Could you add a test case (either a doctest or a real test in `tests/`) that uses an `\xa0` character?

Reply:
would this be a valid test case for that: >>> parse_number('1 099', locale='ru')

Reply:
made some changes applying some of the logic from parse_decimal and it says its passing the test cases on my end